config/routes.rb

```diff
@@ -23,7 +23,11 @@
       patch :update_visitee
     end
 
-    resources :diagnosis, only: %i[index new create show]
+    resources :diagnosis, only: %i[index new create show] do
+      member do
+        get 'step-2' => 'diagnosis#step2'
```

Comment:
You could remove `member` block and add `, on: :member` at the end of the line, but whatever